docs/gguf.md

```diff
@@ -0,0 +1,546 @@
+- `[llm].attention.head_count_kv: uint32`: The number of heads per group used in Grouped-Query-Attention. If not present, the model does not use GQA.
+- `[llm].attention.max_alibi_bias: float32`: The maximum bias to use for ALiBI.
+- `[llm].attention.clamp_kqv: float32`: Value (`C`) to clamp the values of the `Q`, `K`, and `V` tensors between (`[-C, C]`).
+- `[llm].attention.layer_norm_epsilon: float32`: Layer normalization epsilon.
+- `[llm].attention.layer_norm_rms_epsilon: float32`: Layer RMS normalization epsilon.
+
+#### RoPE
+
+- `[llm].rope.dimension_count: uint32`: The number of rotary dimensions for RoPE.
+- `[llm].rope.scale: float32`: A scale factor for RoPE to adjust the context length.
+
+#### Models
+
+The following sections describe the metadata for each model architecture. Each key specified _must_ be present.
+
+##### LLaMA
+
+- `llama.context_length`
+- `llama.embedding_length`
+- `llama.layer_count`
+- `llama.feedforward_length`
+- `llama.rope.dimension_count`
+- `llama.attention.head_count`
+- `llama.attention.layer_norm_rms_epsilon`
+
+###### Optional
+
+- `llama.rope.scale`
+- `llama.attention.head_count_kv`
+- `llama.tensor_data_layout`:
+  - `llama.cpp`:
+    ```python
+    def permute(weights: NDArray, n_head: int) -> NDArray:
+        return (weights.reshape(n_head, 2, weights.shape[0] // n_head // 2, *weights.shape[1:])
+                    .swapaxes(1, 2)
+                    .reshape(weights.shape))
+    ```
+
+##### MPT
+
+- `mpt.context_length`
+- `mpt.embedding_length`
+- `mpt.layer_count`
+- `mpt.attention.head_count`
+- `mpt.attention.alibi_bias_max`
+- `mpt.attention.clip_kqv`
+- `mpt.attention.layer_norm_epsilon`
+
+##### GPT-NeoX
+
+- `gptneox.context_length`
+- `gptneox.embedding_length`
+- `gptneox.layer_count`
+- `gptneox.use_parallel_residual`
+- `gptneox.rope.dimension_count`
+- `gptneox.attention.head_count`
+- `gptneox.attention.layer_norm_epsilon`
+
+###### Optional
+
+- `gptneox.rope.scale`
+
+##### GPT-J
+
+- `gptj.context_length`
+- `gptj.embedding_length`
+- `gptj.layer_count`
+- `gptj.rope.dimension_count`
+- `gptj.attention.head_count`
+- `gptj.attention.layer_norm_epsilon`
+
+###### Optional
+
+- `gptj.rope.scale`
+
+##### GPT-2
+
+- `gpt2.context_length`
+- `gpt2.embedding_length`
+- `gpt2.layer_count`
+- `gpt2.attention.head_count`
+- `gpt2.attention.layer_norm_epsilon`
+
+##### BLOOM
+
+- `bloom.context_length`
+- `bloom.embedding_length`
+- `bloom.layer_count`
+- `bloom.feedforward_length`
+- `bloom.attention.head_count`
+- `bloom.attention.layer_norm_epsilon`
+
+##### Falcon
+
+- `falcon.context_length`
+- `falcon.embedding_length`
+- `falcon.layer_count`
+- `falcon.attention.head_count`
+- `falcon.attention.head_count_kv`
+- `falcon.attention.use_norm`
+- `falcon.attention.layer_norm_epsilon`
+
+###### Optional
+
+- `falcon.tensor_data_layout`:
+
+  - `llama.cpp` (this name may be inaccurate depending on where the Falcon implementation ends up):
```

Comment:
maybe better to change this to reference the creator of the format, in this case `jploski` i guess? https://github.com/jploski/ggml/blob/d5295b477fb36c69468c3fecb0393a8d7980b7c8/examples/falcon/convert-hf-to-ggml.py#L107-L123